### PR TITLE
dialects: Add riscv ecall instruction

### PR DESCRIPTION
**Pull request by JosseVanDelm**
Please note that there is an error in FileCheck. Is this related to #428 ?

### CODE REVIEW

**Review by superlopuh — APPROVED** [review]

File: xdsl/dialects/riscv.py [hunk 1]
```diff
@@ -174,6 +174,27 @@ def __init__(
...
+    This is called S-Type in the RISC-V specification.
+    """
+
+    rd: Annotated[OpResult, RegisterType]
+    rs: Annotated[Operand, RegisterType]
+
+    def __init__(
+        self,
+        rs: Operation | SSAValue,
+    ):
```
Comment (superlopuh) [inline xdsl/dialects/riscv.py:191]:
```suggestion
    ):
    rd = RegisterType(Register())
```

File: xdsl/dialects/riscv.py [hunk 2]
```diff
@@ -174,6 +174,27 @@ def __init__(
...
+        super().__init__(
+            operands=[rd, rs],
```
Comment (superlopuh) [inline xdsl/dialects/riscv.py:193]:
```suggestion
            operands=[rs],
            result_types=[rd],
```

File: xdsl/dialects/riscv.py [hunk 3]
```diff
@@ -704,6 +725,22 @@ class LiOp(RdImmOperation):
     name = "riscv.li"
 
 
+@irdl_op_definition
+class EcallOp(RdRsOperation):
```
Comment (AntonLydike) [inline xdsl/dialects/riscv.py:729]: Should `ecall` have `rd` and `rs`? I think it should be a `NullaryOperation`, right?

**Review by AntonLydike — CHANGES_REQUESTED** [review]
I think there are a few things you should fix here, the pyright thing and `NullaryOperation`.

**Review by AntonLydike — APPROVED** [review]
LGTM

**Pull request merged**